### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by ragrus-nbl**
- Documentation;
- Agent and device components of the app.

### CODE REVIEW

File: src/device/Main.device.nut [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            return;
+        }
+
+        logger.info("Starting flashing. Chunks to flash: " + chunksNum, LOG_SOURCE.APP);
+
+        _chunksToFlash = chunksNum;
+        _expectedChunk = 0;
+
+        local err = 0;
+        if (err = _stm32.connect()           ||
```
Comment (ppetrosh) [inline src/device/Main.device.nut:79]: This is not working as you would expect. This construction doesn't make sense in general. It's a sequence of boolean operators applied to integer values. As a result, err is boolean. And printing it won't give any meaningful information.

Comment (ppetrosh) [reply src/device/Main.device.nut:79]: No, I'm wrong! What the ... trick!

**Comment by ppetrosh** [comment]
I would probably add an HTTP REST endpoint to trigger the flash programming operation. That would be safer and provide a better overall UX.

**Comment by ragrus-nbl** [comment]
> I would probably add an HTTP REST endpoint to trigger the flash programming operation. That would be safer and provide a better overall UX.

Yes, it is already done.

**Pull request merged**